tests/AerospikeCacheTest.php

```diff
@@ -3,7 +3,200 @@
+    }
+
+    /**
+     * @dataProvider provideStatusCodes
+     */
+    public function testShouldClearWithEmptyNamespaceName(int $aerospikeStatusCode, bool $expectedClearSuccessful): void
+    {
+        $aerospikeMock = $this->createMock(\Aerospike::class);
+        $aerospikeCache = new AerospikeCache($aerospikeMock, 'test', 'cache');
+
+        $aerospikeMock->method('truncate')
+            ->willReturn($aerospikeStatusCode);
+
+        $clearSuccessful = $aerospikeCache->clear();
+
+        $this->assertSame($expectedClearSuccessful, $clearSuccessful);
+    }
+
+    public function doClearWithEmptyNamespaceProvider(): array
+    {
+        return [[\Aerospike::OK, true] , [\Aerospike::ERR_CLIENT, false]];
+    }
+
+    /**
+     * @dataProvider provideStatusCodesForClearingNamespace
+     */
+    public function testShouldClearNamespace(int $statusCodeForRemove, int $statusCodeForScan, bool $expectedValue): void
+    {
+        $aerospikeMock = $this->createMock(\Aerospike::class);
+        $aerospikeCache = new AerospikeCache($aerospikeMock, 'test', 'cache', 'testNamespace');
+
+        $aerospikeMock->method('remove')
+            ->willReturn($statusCodeForRemove);
+
+        $aerospikeMock->method('scan')
+            ->willReturnCallback(function ($namespace, $set, $callback) use ($statusCodeForScan) {
+                $callback(['key' => ['key' => 'testNamespace::test']]);
+
+                return $statusCodeForScan;
+            });
+
+        $clearResult = $aerospikeCache->clear();
+
+        $this->assertSame($expectedValue, $clearResult);
+    }
+
+    public function provideStatusCodesForClearingNamespace(): array
+    {
+        return [
+            [\Aerospike::OK, \Aerospike::OK, true],
+            [\Aerospike::OK, \Aerospike::ERR_CLIENT, false],
+            [\Aerospike::ERR_CLIENT, \Aerospike::ERR_CLIENT, false],
+            [\Aerospike::ERR_CLIENT, \Aerospike::OK, false],
+        ];
+    }
+
+    /**
+     * @dataProvider provideStatusCodes
+     */
+    public function testShouldDeleteItem(int $aerospikeStatusCode, bool $expectedDeleteToSucced): void
+    {
+        $aerospikeMock = $this->createMock(\Aerospike::class);
+        $aerospikeCache = new AerospikeCache($aerospikeMock, 'test', 'cache', 'testNamespace');
+
+        $aerospikeMock->method('initKey')
+            ->willReturn(['foo']);
+        $aerospikeMock->method('remove')
+            ->willReturn($aerospikeStatusCode);
+
+        $deleteSuccessful = $aerospikeCache->deleteItem('foo');
+
+        $this->assertSame($expectedDeleteToSucced, $deleteSuccessful);
+    }
+
+    public function provideStatusCodes(): array
+    {
+        return [
+            [\Aerospike::OK, true],
+            [\Aerospike::ERR_CLIENT, false],
+        ];
+    }
+
+    /**
+     * @dataProvider doFetchProvider
+     */
+    public function testShouldReadExistingRecordsFromAerospike(array $cacheItemKeys, array $valuesReturnedByAerospike): void
```

Comment:
This test has quite a lot of logic, could it be simplifed somehow? It almost looks, it could have tests of its own.. There is a key creation logic, callback with aerospike logic and items checking logic.. what is really tested here?

Reply:
I was thinking about it and i will rewrite the test. by using reflection i can test the method directly and remove a lot of logic.

Reply:
Simplified and split into two tests